docs/docs/tutorials/orca-yarn-tutorial.md

```diff
@@ -0,0 +1,475 @@
+
+### 2.1.1 Setup JAVA Environment
+```bash
+# For Ubuntu
+sudo apt-get install openjdk-8-jre
+export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64/
+
+# For CentOS
+su -c "yum install java-1.8.0-openjdk"
+export JAVA_HOME=/usr/lib/jvm/java-1.8.0-openjdk-1.8.0.282.b08-1.el7_9.x86_64/jre
+
+export PATH=$PATH:$JAVA_HOME/bin
+java -version  # Verify the version of JDK.
+```
+
+### 2.1.2 Setup Hadoop Environment
+Check the Hadoop setup and configurations of our cluster. Make sure we correctly set the environment variable `HADOOP_CONF_DIR`, which is needed to initialize Spark on YARN:
+```bash
+export HADOOP_CONF_DIR=/path/to/hadoop/conf
+```
+
+## 2.2 Install Python Libraries
+### 2.2.1 Install Conda
+We need first to use conda to prepare the Python environment on the local machine where we submit our application. You could download and install Conda following [Conda User Guide](https://docs.conda.io/projects/conda/en/latest/user-guide/install/linux.html) or executing the command as below.
+```bash
+# Download Anaconda installation script 
+wget -P /tmp https://repo.anaconda.com/archive/Anaconda3-2020.02-Linux-x86_64.sh
+
+# Execute the script to install conda
+bash /tmp/Anaconda3-2020.02-Linux-x86_64.sh
+
+# Please type this command in your terminal to activate Conda environment
+source ~/.bashrc
+``` 
+
+### 2.2.2 Use Conda to Install BigDL
+Create a conda environment, install BigDL and all the needed Python libraries in the created conda environment:
+``` bash
+# "bigdl" is conda environment name, you can use any name you like.
+# Please change Python version to 3.6 if you need a Python 3.6 environment.
+conda create -n bigdl python=3.7 
+conda activate bigdl
+```
+Alternatively, You can install the latest release version of BigDL (built on top of Spark 2.4.6 by default) as follows:
+```bash
+pip install bigdl
+```
+You can install the latest nightly build of BigDL as follows:
+```bash
+pip install --pre --upgrade bigdl
+```
+
+__Note:__ 
+* Using Conda to install BigDL will automatically install libraries including `conda-pack`, `torch`, `torchmetrics`, `torchvision`, `pandas`, `numpy`, `pyspark==2.4.6`, and etc.
+* You can install BigDL built on top of Spark 3.1.2 as follows:
+    
+    ```bash
+    # Install the latest release version
+    pip install bigdl-spark3
+
+    # Install the latest nightly build version
+    pip install --pre --upgrade bigdl-spark3 
+    ```
+* Installing bigdl-spark3 will automatically install `pyspark==3.1.2`.
+
+Please see more details in [Python User Guide](https://bigdl.readthedocs.io/en/latest/doc/UserGuide/python.html).
+
+## 2.3 For CDH Users
+If the __client node__ where you submit applications has installed Spark, it might conflict with the `pyspark` installed automatically with BigDL on the __client node__ if they versions not match. Please follow the steps below to solve the conflict.
```

Comment:
* For CDH user, the value of environment variable `HADOOP_CONF_DIR` should be `/etc/hadoop/conf` by default

* The client node (where you submit the program to the YARN cluster) may have already installed a different version of Spark than the one installed with BigDL; to avoid conflicts, unset all Spark-related environment variables (you may use use `env | grep SPARK` to find all of them):
   ```bash
   unset SPARK_HOME
   unset SPARK_VERSION
   unset ...
   ```
* You also need to properly export the location of `libhdfs.so` if needed. For instance, when using `pyarrow`, set up the environment variable `ARROW_LIBHDFS_DIR` on the client node (where you submits the program to the YARN cluster) as follows:
   ```bash
   export ARROW_LIBHDFS_DIR=/opt/cloudera/parcels/CDH-${CHD_VERSION}/lib64/
   ```